### PR TITLE
fix/PSD-3985_missing_app_security_headers

### PR DESCRIPTION
**Pull request by tahir-khalid**
## Description

PSD-3985

Missing or Misconfigured Application Security Headers - A number of HTTP headers which can be used to enhance the security posture of the web application were not used.

Implemented CSP and HSTS support in the rails app.

### CODE REVIEW

File: config/initializers/content_security_policy.rb [hunk 1]
```diff
@@ -12,57 +12,35 @@
...
-
-#   # Generate session nonces for permitted importmap, inline scripts and inline styles
-#   config.content_security_policy_nonce_generator = ->(request) { request.session.id.to_s }
-#   config.content_security_policy_nonce_directives = %w[script-src style-src]
-
-#   # For now, only report to any reports to Sentry in production
-#   config.content_security_policy_report_only = true
-# end
+# Enable CSP nonce generation
+Rails.application.config.content_security_policy_nonce_generator = ->(_request) { SecureRandom.base64(16) }
```
[the remark this answers is not in the document]
Comment (tahir-khalid) [reply config/initializers/content_security_policy.rb:34]: Confirmed no other changes needed.

**Comment by tahir-khalid** [comment]
> Looks ok, I would like to know how this is going to be tested and verified please. How would you get Venu to test for example?

@alan-at-work I have updated the QA ticket with instructions (https://regulatorydelivery.atlassian.net/browse/PSD-3988) that should be adequate for Venu to test.

**Review by alan-at-work — APPROVED** [review]

File: config/initializers/content_security_policy.rb [hunk 2]
```diff
@@ -12,57 +12,24 @@
...
+
+Rails.application.config.content_security_policy do |policy|
+  policy.default_src :self
+  policy.object_src :none
+  policy.child_src :self
+  policy.frame_ancestors :none
+  policy.upgrade_insecure_requests true
+  policy.block_all_mixed_content true
+  policy.script_src(*trusted_script_sources)
+  policy.style_src(*trusted_style_sources)
```
Comment (meeravnagesh) [inline config/initializers/content_security_policy.rb:28]: One question , I may not be asking relevant but we use lots of components from Gov UK designs , so when we say trusted font and trusted source  is just google... , will it not affect the gov uk components ?

i might be wrong :) let me know

**Review by meeravnagesh — CHANGES_REQUESTED** [review]

File: config/initializers/content_security_policy.rb (hunk 2 above)
Comment (tahir-khalid) [reply config/initializers/content_security_policy.rb:33]: Updated CSP entries to match the latest GOV UK Design requirements.

Comment (meeravnagesh) [reply config/initializers/content_security_policy.rb:28]: we have added gov.uk pattern to the trusted sites

**Review by meeravnagesh — APPROVED** [review]

**Pull request merged**